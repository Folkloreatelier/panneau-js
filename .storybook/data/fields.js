export default [
    {
        type: 'test',
        name: 'a-test',
        label: 'A Test',
        placeholder: 'cool',
    },
    {
        type: 'test',
        name: 'a-test-h',
        label: 'A test H',
        horizontal: true,
    },
    {
        type: 'fields',
        name: 'cool-row',
        row: true,
        label: 'Cool row',
        fields: [
            {
                name: 'b-test',
                type: 'test',
                placeholder: 'b-test',
            },
            {
                name: 'c-test',
                type: 'test',
                placeholder: 'c-test',
            },
            {
                name: 'd-test',
                type: 'test',
                placeholder: 'd-test',
            },
        ],
    },
    {
        type: 'fields',
        name: 'Cooool hhh',
        label: 'Cool H',
        row: true,
        fields: [
            {
                type: 'test',
                name: 'e-test',
                label: 'e-test',
            },
            {
                type: 'test',
                name: 'f-test',
                label: 'f-test',
            },
        ],
    },
];
