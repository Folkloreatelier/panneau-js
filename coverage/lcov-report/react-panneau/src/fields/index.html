<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-panneau/src/fields</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> react-panneau/src/fields
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.15% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>105/187</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.68% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>48/118</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.26% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>21/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.66% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>105/179</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="CodeField.jsx"><a href="CodeField.jsx.html">CodeField.jsx</a></td>
	<td data-value="71.05" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	<td data-value="39.29" class="pct low">39.29%</td>
	<td data-value="28" class="abs low">11/28</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	</tr>

<tr>
	<td class="file low" data-value="ColorField.jsx"><a href="ColorField.jsx.html">ColorField.jsx</a></td>
	<td data-value="48.21" class="pic low"><div class="chart"><div class="cover-fill" style="width: 48%;"></div><div class="cover-empty" style="width:52%;"></div></div></td>
	<td data-value="48.21" class="pct low">48.21%</td>
	<td data-value="56" class="abs low">27/56</td>
	<td data-value="28" class="pct low">28%</td>
	<td data-value="25" class="abs low">7/25</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	<td data-value="48.21" class="pct low">48.21%</td>
	<td data-value="56" class="abs low">27/56</td>
	</tr>

<tr>
	<td class="file high" data-value="SelectField.jsx"><a href="SelectField.jsx.html">SelectField.jsx</a></td>
	<td data-value="80" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="25" class="abs high">20/25</td>
	<td data-value="62.96" class="pct medium">62.96%</td>
	<td data-value="27" class="abs medium">17/27</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="25" class="abs high">20/25</td>
	</tr>

<tr>
	<td class="file medium" data-value="TextField.jsx"><a href="TextField.jsx.html">TextField.jsx</a></td>
	<td data-value="58.49" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 58%;"></div><div class="cover-empty" style="width:42%;"></div></div></td>
	<td data-value="58.49" class="pct medium">58.49%</td>
	<td data-value="53" class="abs medium">31/53</td>
	<td data-value="38.24" class="pct low">38.24%</td>
	<td data-value="34" class="abs low">13/34</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="58.49" class="pct medium">58.49%</td>
	<td data-value="53" class="abs medium">31/53</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Sep 05 2017 22:28:34 GMT-0400 (EDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
