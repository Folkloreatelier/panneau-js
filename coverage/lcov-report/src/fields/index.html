<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/fields</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> src/fields
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.3% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>71/855</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.14% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>28/344</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">9.6% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/125</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.08% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>71/543</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CodeField.jsx"><a href="CodeField.jsx.html">CodeField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="97" class="abs low">0/97</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="38" class="abs low">0/38</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="20" class="abs low">0/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="52" class="abs low">0/52</td>
	</tr>

<tr>
	<td class="file low" data-value="ColorField.jsx"><a href="ColorField.jsx.html">ColorField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="119" class="abs low">0/119</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="75" class="abs low">0/75</td>
	</tr>

<tr>
	<td class="file low" data-value="DateField.jsx"><a href="DateField.jsx.html">DateField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="188" class="abs low">0/188</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="82" class="abs low">0/82</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="34" class="abs low">0/34</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="117" class="abs low">0/117</td>
	</tr>

<tr>
	<td class="file low" data-value="LocaleField.jsx"><a href="LocaleField.jsx.html">LocaleField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="95" class="abs low">0/95</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="54" class="abs low">0/54</td>
	</tr>

<tr>
	<td class="file medium" data-value="SelectField.jsx"><a href="SelectField.jsx.html">SelectField.jsx</a></td>
	<td data-value="51.39" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 51%;"></div><div class="cover-empty" style="width:49%;"></div></div></td>
	<td data-value="51.39" class="pct medium">51.39%</td>
	<td data-value="72" class="abs medium">37/72</td>
	<td data-value="24.59" class="pct low">24.59%</td>
	<td data-value="61" class="abs low">15/61</td>
	<td data-value="41.67" class="pct low">41.67%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="52.11" class="pct medium">52.11%</td>
	<td data-value="71" class="abs medium">37/71</td>
	</tr>

<tr>
	<td class="file low" data-value="SliderField.jsx"><a href="SliderField.jsx.html">SliderField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="61" class="abs low">0/61</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="21" class="abs low">0/21</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="34" class="abs low">0/34</td>
	</tr>

<tr>
	<td class="file medium" data-value="TextField.jsx"><a href="TextField.jsx.html">TextField.jsx</a></td>
	<td data-value="56.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 56%;"></div><div class="cover-empty" style="width:44%;"></div></div></td>
	<td data-value="56.67" class="pct medium">56.67%</td>
	<td data-value="60" class="abs medium">34/60</td>
	<td data-value="34.21" class="pct low">34.21%</td>
	<td data-value="38" class="abs low">13/38</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="14" class="abs medium">7/14</td>
	<td data-value="56.67" class="pct medium">56.67%</td>
	<td data-value="60" class="abs medium">34/60</td>
	</tr>

<tr>
	<td class="file low" data-value="TextLocaleField.jsx"><a href="TextLocaleField.jsx.html">TextLocaleField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	</tr>

<tr>
	<td class="file low" data-value="ToggleField.jsx"><a href="ToggleField.jsx.html">ToggleField.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="97" class="abs low">0/97</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="34" class="abs low">0/34</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="47" class="abs low">0/47</td>
	</tr>

<tr>
	<td class="file low" data-value="collection.js"><a href="collection.js.html">collection.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="30" class="abs low">0/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="17" class="abs low">0/17</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 28 2017 13:57:59 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
