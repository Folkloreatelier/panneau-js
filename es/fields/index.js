import Select from './SelectField';
import Text from './TextField';
import Color from './ColorField';
import Code from './CodeField';
import Slider from './SliderField';
import Date from './DateField';
import fieldsCollection from './collection';

export { Select, Text, Color, Code, Slider, Date, fieldsCollection };