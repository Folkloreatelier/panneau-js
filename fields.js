module.exports = require('./lib/fields/index'); // eslint-disable-line global-require
