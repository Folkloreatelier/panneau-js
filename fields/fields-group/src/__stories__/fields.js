import { ComponentsCollection } from '@panneau/core';

import Text, { TextLocale } from '../../../text/src/index';
import Code from '../../../code/src/index';
import Select from '../../../select/src/index';
import Color from '../../../color/src/index';
import Date from '../../../date/src/index';
import Slider from '../../../slider/src/index';
import Toggle from '../../../toggle/src/index';

const fieldsCollection = new ComponentsCollection({
    Text,
    TextLocale,
    Code,
    Select,
    Color,
    Date,
    Slider,
    Toggle,
});

export default fieldsCollection;
