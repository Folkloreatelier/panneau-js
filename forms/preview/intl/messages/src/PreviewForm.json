[
  {
    "id": "forms.normal.buttons.submit",
    "description": "The label of the \"save\" form button",
    "defaultMessage": "Save"
  }
]