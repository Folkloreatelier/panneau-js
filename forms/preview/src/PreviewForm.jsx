/* eslint-disable react/button-has-type */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import classNames from 'classnames';
import SplitPane from 'react-split-pane';
import {
    PropTypes as PanneauPropTypes,
    withPreviewsCollection,
    withFormsCollection,
} from '@panneau/core';

import styles from './styles.scss';

const stylePropType = PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]));

const propTypes = {
    formsCollection: PanneauPropTypes.componentsCollection.isRequired,
    previewsCollection: PanneauPropTypes.componentsCollection.isRequired,
    value: PropTypes.object, // eslint-disable-line react/forbid-prop-types
    form: PropTypes.string,
    preview: PropTypes.string,
    paneStyle: stylePropType,
    paneSplit: PropTypes.oneOf(['vertical', 'horizontal']),
    paneStep: PropTypes.number,
    panesStyle: stylePropType,
    paneFormStyle: stylePropType,
    panePreviewStyle: stylePropType,
    paneResizerStyle: stylePropType,
    formWidth: PropTypes.number,
    formMinWidth: PropTypes.number,
    formMaxWidth: PropTypes.number,
    formStyle: stylePropType,
    previewStyle: stylePropType,
};

const defaultProps = {
    value: null,
    form: 'normal',
    preview: 'json',
    paneStyle: { position: 'relative' },
    paneSplit: 'vertical',
    paneStep: 1,
    panesStyle: null,
    paneFormStyle: null,
    panePreviewStyle: null,
    paneResizerStyle: null,
    formWidth: 300,
    formMinWidth: 200,
    formMaxWidth: null,
    formStyle: null,
    previewStyle: null,
};

// eslint-disable-next-line react/prefer-stateless-function
export class PreviewForm extends Component {
    render() {
        const {
            formsCollection,
            previewsCollection,
            value,
            form,
            preview,
            paneStyle,
            paneSplit,
            paneStep,
            panesStyle,
            paneFormStyle,
            panePreviewStyle,
            paneResizerStyle,
            formWidth,
            formMinWidth,
            formMaxWidth,
            formStyle,
            previewStyle,
            ...props
        } = this.props;

        const FormComponent = formsCollection.getComponent(form);
        const PreviewComponent = previewsCollection.getComponent(preview);

        return (
            <div className={styles.container}>
                <SplitPane
                    split={paneSplit}
                    step={paneStep}
                    defaultSize={formWidth}
                    minSize={formMinWidth}
                    maxSize={formMaxWidth}
                    style={paneStyle}
                    paneStyle={panesStyle}
                    pane1Style={paneFormStyle}
                    pane2Style={panePreviewStyle}
                    resizerStyle={paneResizerStyle}
                >
                    <div className={styles.form}>
                        {FormComponent !== null ? <FormComponent value={value} {...props} /> : null}
                    </div>
                    <div className={styles.preview}>
                        {PreviewComponent !== null ? (
                            <PreviewComponent value={value} {...props} />
                        ) : null}
                    </div>
                </SplitPane>
            </div>
        );
    }
}

PreviewForm.propTypes = propTypes;
PreviewForm.defaultProps = defaultProps;

const WithPreviewCollectionContainer = withPreviewsCollection()(PreviewForm);
const WithFormsCollectionContainer = withFormsCollection()(WithPreviewCollectionContainer);
export default WithFormsCollectionContainer;
