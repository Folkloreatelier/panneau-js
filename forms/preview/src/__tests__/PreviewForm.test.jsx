import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import { withFieldsCollection } from '@panneau/fields';
/* eslint-enable import/no-extraneous-dependencies */
import { IntlProvider } from 'react-intl';
import PreviewForm from '../PreviewForm';

const PreviewFormWithFields = withFieldsCollection({
    childContext: true,
})(PreviewForm);

const PreviewFormWithForms = withFormsCollection({
    childContext: true,
})(PreviewFormWithFields);

const PreviewFormWithPreviews = withPreviewsCollection({
    childContext: true,
})(PreviewFormWithForms);

const fields = [
    {
        type: 'text',
        label: 'Text',
        name: 'text',
    },
];

const value = {
    text: 'Text',
};

test('match snapshot', () => {
    const component = renderer.create((
        <IntlProvider locale="en">
            <PreviewFormWithPreviews
                fields={fields}
                value={value}
            />
        </IntlProvider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
