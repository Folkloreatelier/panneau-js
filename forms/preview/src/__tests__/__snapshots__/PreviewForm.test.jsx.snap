// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className=""
>
  <div
    className="SplitPane  vertical "
    style={
      Object {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "left": 0,
        "msUserSelect": "text",
        "outline": "none",
        "overflow": "hidden",
        "position": "relative",
        "right": 0,
        "userSelect": "text",
      }
    }
  >
    <div
      className="Pane vertical Pane1  "
      style={
        Object {
          "flex": "none",
          "outline": "none",
          "position": "relative",
          "width": 300,
        }
      }
    >
      <div>
        <div>
          <form
            action=""
            method="POST"
            onSubmit={[Function]}
          >
            <div>
              <div>
                <div>
                  <div
                    className="form-group form-group-text"
                  >
                    <div
                      className="form-group-label"
                    >
                      <label
                        className="control-label"
                        htmlFor="text"
                      >
                        Text
                      </label>
                    </div>
                    <div
                      className="form-group-inner"
                      style={
                        Object {
                          "display": "block",
                        }
                      }
                    >
                      <input
                        className="form-control field-text"
                        disabled={false}
                        id="text"
                        maxLength={null}
                        name="text"
                        onBlur={null}
                        onChange={[Function]}
                        onFocus={null}
                        placeholder={null}
                        type="text"
                        value="Text"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="mt-4 border-top pt-2"
            >
              <div
                className="row align-items-center"
              >
                <div
                  className="col-sm col-sm-auto text-right"
                >
                  <div
                    className=""
                  >
                    <div
                      className="btn-group"
                    >
                      <button
                        className="btn btn-primary"
                        onClick={null}
                        type="submit"
                      >
                        <span>
                          Save
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  className="col-sm order-sm-first text-right"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <span
      className="Resizer vertical "
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="presentation"
      style={Object {}}
    />
    <div
      className="Pane vertical Pane2  "
      style={
        Object {
          "flex": 1,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div>
        <pre>
          {
    "text": "Text"
}
        </pre>
      </div>
    </div>
  </div>
</div>
`;
